plugin/documents.py

```diff
@@ -492,9 +504,35 @@ def _when_selection_remains_stable_async(self, f: Callable[[], None], r: sublime
 
     def _register_async(self) -> None:
         file_name = self.view.file_name()
-        if file_name:
-            self._file_name = file_name
-            self.manager.register_listener_async(self)
+        if not file_name:
+            debug("listener", self, "has  no file name")
```

Comment:
The `listener` word is redundant now that `__rep__` is more descriptive.
Similarly, in places below I guess.